x/wasm/keeper/keeper.go

```diff
@@ -195,6 +197,80 @@ func (k Keeper) GetStoreKey() sdk.StoreKey {
 	return k.storeKey
 }
 
+func (k Keeper) IsContractMethodBlocked(ctx sdk.Context, contractAddr, method string) bool {
+	blockedMethods := k.GetContractMethodBlockedList(ctx, contractAddr)
+	return blockedMethods.IsMethodBlocked(method)
+}
+
+func (k Keeper) GetContractMethodBlockedList(ctx sdk.Context, contractAddr string) *types.ContractMethods {
+	if ctx.UseParamCache() {
+		if GetWasmParamsCache().IsNeedBlockedUpdate() {
+			cms := k.getAllBlockedList(ctx)
+			if !ctx.IsCheckTx() {
+				GetWasmParamsCache().UpdateBlockedContractMethod(cms)
+			}
+			return types.FindContractMethods(cms, contractAddr)
+		}
+		return GetWasmParamsCache().GetBlockedContractMethod(contractAddr)
+	}
+
+	return k.getContractMethodBlockedList(ctx, contractAddr)
+}
+
+func (k Keeper) getAllBlockedList(ctx sdk.Context) []*types.ContractMethods {
+	store := k.ada.NewStore(ctx.GasMeter(), ctx.KVStore(k.storeKey), nil)
+	it := sdk.KVStorePrefixIterator(store, types.GetContractMethodBlockedListPrefix(""))
+	defer it.Close()
+
+	var cms []*types.ContractMethods
+	for ; it.Valid(); it.Next() {
+		var method types.ContractMethods
+		err := proto.Unmarshal(it.Value(), &method)
+		if err != nil {
+			k.Logger(ctx).Error("getAllBlockedList", "unmarshal error", err)
+			continue
+		}
+		cms = append(cms, &method)
+	}
+	return cms
+}
+
+func (k Keeper) getContractMethodBlockedList(ctx sdk.Context, contractAddr string) *types.ContractMethods {
+	store := k.ada.NewStore(ctx.GasMeter(), ctx.KVStore(k.storeKey), nil)
+	key := types.GetContractMethodBlockedListPrefix(contractAddr)
+	data := store.Get(key)
+	var blockedMethods types.ContractMethods
+	err := proto.Unmarshal(data, &blockedMethods)
+	if err != nil {
+		k.Logger(ctx).Error("getContractMethodBlockedList", "unmarshal error", err)
```

Comment:
I think unmarshal err must panic. if len == 0 can return empty obj

Reply:
panic is ok.
proto.Unmarshal() will return nil when len(data) == 0